### PR TITLE
[ci] add a flag to upload/not-upload docker images

### PR DESCRIPTION
**Pull request by can-anyscale**
Add a flag to upload or not-upload docker images. This is required to build ray images for some tests that doesn't need to upload the images.

Test:
- CI

### CODE REVIEW

**Review by aslonnie — COMMENTED** [review]
this seems to work only when uploading to ecr?

**Comment by can-anyscale** [comment]
Should work for docker hub, where ray image is stored as well ;)

File: ci/ray_ci/ray_docker_container.py [hunk 1]
```diff
@@ -56,7 +56,10 @@ def run(self) -> None:
         self.run_script(cmds)
 
     def _should_upload(self) -> bool:
-        return os.environ.get("BUILDKITE_PIPELINE_ID") == POSTMERGE_PIPELINE
+        return (
```
Comment (can-anyscale) [inline ci/ray_ci/ray_docker_container.py:59]: this check is for dockerhub uploading

File: ci/ray_ci/anyscale_docker_container.py [hunk 2]
```diff
@@ -23,15 +23,19 @@ def run(self) -> None:
...
         # tag
-        for alias in self._get_image_tags():
-            alias_image = f"{ecr}/anyscale/{self.image_type}:{alias}"
-            cmds += [
-                f"docker tag {anyscale_image} {alias_image}",
-                f"docker push {alias_image}",
-            ]
+        if self._should_upload():
+            for alias in self._get_image_tags():
+                alias_image = f"{ecr}/anyscale/{self.image_type}:{alias}"
```
Comment (aslonnie) [inline ci/ray_ci/anyscale_docker_container.py:28]: if it is docker hub, then there needs to be no slash after the ecr?

**Review by aslonnie — COMMENTED** [review]
also, there is an unresolved conflict against master head.

**Comment by can-anyscale** [comment]
@aslonnie resolve conflict, this `--upload` flag is defined on the container base class so it works for both ray_docker_container (which uploads to dockerhub) and anyscale_docker_container (which uploads to ecr)

**Comment by aslonnie** [comment]
> @aslonnie resolve conflict, this `--upload` flag is defined on the container base class so it works for both ray_docker_container (which uploads to dockerhub) and anyscale_docker_container (which uploads to ecr)

for uploading to dockerhub, what is the `ecr` being set?

**Review by aslonnie — APPROVED** [review]

**Comment by can-anyscale** [comment]
Discuss this offlined

**Pull request merged**